crates/core/src/routing_info/mod.rs

```diff
@@ -37,6 +41,13 @@ pub enum Command {
     SyncRoutingInformation,
 }
 
+pub enum PartitionAuthoritativeNode {
+    /// The local node is an authoritative source for this partition.
+    Local,
+    /// The partition resides on a remote node.
+    Remote(NodeId),
+}
```

Comment:
I’m curious if the distinction between local and remote is something the `PartitionRouting` should handle. For instance, if the caller doesn’t care if it’s local or remote, but just wants to know the node_id, then the `Local` variant would need an extra lookup. This is for example the case in the ingress. So are we maybe overfitting the `PartitionRouting` table a bit too much to the `RemoteScannerManager`?

Reply:
I can sense the same smell but I couldn't see an elegant way to solve it. I'll come back to this and give it another try.